.github/workflows/cd.yml

```diff
@@ -0,0 +1,118 @@
+name: Release
+on:
+  push:
+    tags:
+      - 'v*'
+jobs:
+  build_and_upload_artifact_for_macos_and_linux:
+    strategy:
+      matrix:
+        config:
+        - { os: ubuntu-latest, name: 'linux'}
+        - { os: macos-latest, name: 'macos'}
+    runs-on: ${{ matrix.config.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Fetch tag git metadata
+        run: git fetch --force --update-shallow --depth 1 origin 'refs/tags/*:refs/tags/*'
+      - uses: actions/setup-dotnet@v1.8.1
+        with:
+          dotnet-version: 5.0.x
+      - name: Configure cmake
+        run: |
+          mkdir cmake-build-release && cd cmake-build-release && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release ..
+      - name: Build bufc
+        run: |
+          make bufc
+        working-directory: cmake-build-release
+      - name: Update Nuget Source
+        run: |
+          dotnet nuget update source GitHub-StirlingLabs --source https://nuget.pkg.github.com/StirlingLabs/index.json --username ${{github.repository_owner}}  --password ${{github.token}} --store-password-in-clear-text
```

Comment:
Should probably use `StirlingLabs/GithubNugetAuthAction` here instead.

Reply:
Good catch